File: Geeks For Geeks/Linked Lists/Reverse/reverse.py
class Node(object):
	"""docstring for Node"""
	def __init__(self, data):
		self.data = data
		self.next = None

class LinkedList(object):
	"""docstring for LnkedList"""
	def __init__(self):
		self.head = None
		
	def push(self, new_data):
		new_node = Node(new_data)
		new_node.next = self.head
		self.head = new_node

	def reverse(self):
		# 1>2>3>4
		# 4>3>2>1
		prev = None
		current = self.head
		if current is None:
			return
		while current:
			temp = current.next
			current.next = prev
			prev = current
			current = temp
		self.head = prev
	
	def printList(self):
		h = self.head
		while(h):
			print h.data
			h = h.next

llist = LinkedList()
llist.push(20)
llist.push(4)
llist.push(15)
llist.push(85)
 
print "Given Linked List"
llist.printList()
llist.reverse()
print "\nReversed Linked List"
llist.printList()

File: Geeks For Geeks/Linked Lists/Reverse/reverse.java
class LinkedList{
	static Node head;
	
	static class Node{
		int data;
		Node next;

		Node (int d){
			data = d;
			next = null;
		}
	}

	Node reverse(Node node){

		Node prev = null;
		Node curr = node;
		Node next = null;

		while(curr!=null){
			next = curr.next;
			curr.next = prev;
			prev = curr;
			curr = curr.next;
		}
		node = prev;
		return node;
	}

	void printList(Node node){
		while(node!=null){
			System.out.println(node.data + " ");
			node = node.next;
		}
	}
	public static void main(String[] args){
		LinkedList list = new LinkedList();
		list.head = new Node(85);
		list.head.next = new Node(15);
list.printList(head);
	}

}